### ISSUE TITLE
cosmrs 0.4.0: "failed to resolve: could not find `v1` in `wasm`"

### ISSUE THREAD

**Issue by maxcountryman** [cosmos/cosmos-rust#166]
It seems like the version of `proto` shipped with the crate is out of sync with what's expected. I'm seeing various errors like below, for example:

```
error[E0433]: failed to resolve: could not find `v1` in `wasm`
   --> /Users/max/.cargo/registry/src/github.com-1ecc6299db9ec823/cosmrs-0.4.0/src/cosmwasm.rs:354:36
    |
354 |         proto::cosmwasm::wasm::v1::MsgClearAdmin::from(&msg)
    |                                    ^^^^^^^^^^^^^ not found in `proto::cosmwasm::wasm::v1`
    |
help: consider importing one of these items
    |
6   | use cosmos_sdk_proto::cosmwasm::wasm::v1beta1::MsgClearAdmin;
    |
6   | use crate::proto::cosmwasm::wasm::v1beta1::MsgClearAdmin;
```

**Comment by maxcountryman** [comment]
This seems to have happened because the `cosmos-sdk-proto` crate is included via a `path`; this is hiding the fact that the crate will reference the published version elsewhere, but not when built locally. To reproduce, you can apply this diff:

```diff
diff --git a/cosmrs/Cargo.toml b/cosmrs/Cargo.toml
index bbfa8ee..d3b75d8 100644
--- a/cosmrs/Cargo.toml
+++ b/cosmrs/Cargo.toml
@@ -12,7 +12,7 @@ edition = "2021"
 rust-version = "1.56"

 [dependencies]
-cosmos-sdk-proto = { version = "0.8", default-features = false, path = "../cosmos-sdk-proto" }
+cosmos-sdk-proto = { version = "0.8", default-features = false }
 ecdsa = { version = "0.13", features = ["std"] }
 eyre = "0.6"
 k256 = { version = "0.10", features = ["ecdsa", "sha256"] }
```

And then try to run the tests.

It might be worth decoupling the publishing process for these two crates to prevent this in the future.

**Comment by tony-iqlusion** [comment]
I agree with your assessment. PR to release `cosmos-sdk-proto` v0.9 here:

https://github.com/cosmos/cosmos-rust/pull/167

**Comment by maxcountryman** [comment]
I'd recommend not relying on `path` when publishing, otherwise it's difficult to detect these discrepancies between published and unpublished changes in deps.

**Comment by tony-iqlusion** [comment]
> I'd recommend not relying on path when publishing

What are you suggesting exactly? Removing it to publish, then always following up with another commit to add it back? Seems tedious.

**Comment by tony-iqlusion** [comment]
Fixed by #168
